Fallback to default icon for symlinks

stsewd opened the pull request:
fixes #80 

I mean, it fixes my problem, but maybe some users still want to have an icon for symlinks? Feel free to close the issue if you consider

kyazdani42 commented:
not sure symlinks should have icons. Let see what other people say.

Akin909 commented:
Tbh I think an icon would be nice, so you can see at a glance that an node is a link. I use this in quite nested repos and it's not always clear when the node is indented a lot that is a symlink. I think some kind of unique symbol for that would be good

kyazdani42 commented on lua/lib/renderer.lua:
```diff
@@ -195,9 +195,14 @@ local function update_draw_data(tree, depth, markers)
+      local icon = ""
+      if not icon_state.show_file_icon then
+        icon = icon_state.icons.default
+      else
```
this would be bad for performance, could you make a function that is generated at first (like i did for folders and default icons). Although there shouldn't be too many symlinks, there actually could be (open vim in `/usr/lib` for instance),

stsewd replied:
I'll try to see how to do this. BTW, shouldn't this be another option? Having an icon for symlinks only, not use the same for files.

kyazdani42 replied:
we could add this indeed. If you feel like it :)

stsewd commented:
I have updated this PR to include an `symlink` option for icons, it defaults to the `default` icon. Also looks like this didn't fix my previous problem https://github.com/kyazdani42/nvim-tree.lua/commit/505d63a3e7253a78fcc3c94d9da415fde89e9887 (sorry was using another branch), so I fixed that here as well. Tested this with icons as well, and it looks correct.

kyazdani42 commented on lua/lib/renderer.lua:
```diff
@@ -48,6 +48,21 @@ if icon_state.show_file_icon then
+if icon_state.show_file_icon then
+  get_special_icon = function()
+    -- Special files don't have an icon
+    return " "
```
so they shouldn't return a space ? Please reimplement the previous logic because some people want to display a default icon for each file, even special ones.

stsewd replied:
Fixed

kyazdani42 commented on lua/lib/renderer.lua:
```diff
@@ -48,6 +48,21 @@ if icon_state.show_file_icon then
 
 end
 
+local get_symlink_icon = function() return icon_state.icons.symlink end
```
this should return an empty string.

stsewd replied:
This is the same logic as https://github.com/kyazdani42/nvim-tree.lua/blob/505d63a3e7253a78fcc3c94d9da415fde89e9887/lua/lib/renderer.lua#L32-L32, otherwise the default icon won't take effect when not using icons

kyazdani42 replied:
mmhh indeed, but that should not be expected behaviour. If a user doesn't want icons, we shouldn't display icons at all right ?

stsewd replied:
They shouldn't get any icons from web-icons, and the defaults for icon.default and icon.symlink are empty. My main reason for defaulting to this value is to fix the padding when not using icons, this can be another setting (padding), but this approach is simple enough and it works

kyazdani42 commented on lua/lib/renderer.lua:
```diff
@@ -48,6 +48,21 @@ if icon_state.show_file_icon then
+  end
+end
+
+local get_special_icon = function() return icon_state.icons.default end
```
and this too.

stsewd replied:
Same as https://github.com/kyazdani42/nvim-tree.lua/pull/82#discussion_r466078298

Akin909 commented on doc/nvim-tree-lua.txt:
```diff
@@ -87,11 +87,16 @@ is installed and in your |runtimepath|
 >
   let g:lua_tree_icons = {
       \ 'default': '',
+      \ 'symlink': '🔗',
```
Here are some examples from the [devicons website](https://www.nerdfonts.com/cheat-sheet) - ``.

stsewd replied:
You are right, an icon from nerdfont looks better. Changed
